src/Symfony/Component/Config/Definition/EnumNode.php

```diff
@@ -78,13 +104,50 @@ protected function finalizeValue(mixed $value): mixed
     {
         $value = parent::finalizeValue($value);
 
-        if (!\in_array($value, $this->values, true)) {
-            $ex = new InvalidConfigurationException(\sprintf('The value %s is not allowed for path "%s". Permissible values: %s', json_encode($value), $this->getPath(), $this->getPermissibleValues(', ')));
-            $ex->setPath($this->getPath());
+        if ($this->enumFqcn) {
+            if (\is_a($this->enumFqcn, \BackedEnum::class, true)) {
+                if ($value instanceof \UnitEnum && !$value instanceof \BackedEnum) {
```

Comment:
why checking BackedEnum here and not `$this->enumFqcn` ?

Reply:
and why not providing the dedicated error message for all enums as well ?

Reply:
I updated this part [here](https://github.com/symfony/symfony/compare/4ea6e1e59021a4c3a8906fb7466a5986a53da42e..2511fc917de0a1aaf52c72983376d75e223d633d), is it what you had in mind?